In [3]:
import pandas as pd
import numpy as np
from scipy import stats
from matplotlib import pyplot as plt
import math
import seaborn as sns


Bad key "text.kerning_factor" on line 4 in
C:\Users\britn\anaconda3\envs\iod\lib\site-packages\matplotlib\mpl-data\stylelib\_classic_test_patch.mplstyle.
You probably need to get an updated matplotlibrc file from
https://github.com/matplotlib/matplotlib/blob/v3.1.3/matplotlibrc.template
or from the matplotlib source distribution


In [4]:
df = pd.read_csv('../DATA/avocado.csv')

In [9]:
df

Date  AveragePrice  Total Volume     4046       4225  \
Unnamed: 0                                                               
0           2015-12-27          1.33      64236.62  1036.74   54454.85   
1           2015-12-20          1.35      54876.98   674.28   44638.81   
2           2015-12-13          0.93     118220.22   794.70  109149.67   
3           2015-12-06          1.08      78992.15  1132.00   71976.41   
4           2015-11-29          1.28      51039.60   941.48   43838.39   
...                ...           ...           ...      ...        ...   
7           2018-02-04          1.63      17074.83  2046.96    1529.20   
8           2018-01-28          1.71      13888.04  1191.70    3431.50   
9           2018-01-21          1.87      13766.76  1191.92    2452.79   
10          2018-01-14          1.93      16205.22  1527.63    2981.04   
11          2018-01-07          1.62      17489.58  2894.77    2356.13   

              4770  Total Bags  Small Bags  Large Bags  XLarge Bags  \
Unnamed: 0                                                            
0            48.16     8696.87     8603.62       93.25          0.0   
1            58.33     9505.56     9408.07       97.49          0.0   
2           130.50     8145.35     8042.21      103.14          0.0   
3            72.58     5811.16     5677.40      133.76          0.0   
4            75.78     6183.95     5986.26      197.69          0.0   
...            ...         ...         ...         ...          ...   
7             0.00    13498.67    13066.82      431.85          0.0   
8             0.00     9264.84     8940.04      324.80          0.0   
9           727.94     9394.11     9351.80       42.31          0.0   
10          727.01    10969.54    10919.54       50.00          0.0   
11          224.53    12014.15    11988.14       26.01          0.0   

                    type  year            region  
Unnamed: 0                                        
0           conventional  2015            Albany  
1           conventional  2015            Albany  
2           conventional  2015            Albany  
3           conventional  2015            Albany  
4           conventional  2015            Albany  
...                  ...   ...               ...  
7                organic  2018  WestTexNewMexico  
8                organic  2018  WestTexNewMexico  
9                organic  2018  WestTexNewMexico  
10               organic  2018  WestTexNewMexico  
11               organic  2018  WestTexNewMexico  

[18249 rows x 13 columns]

In [6]:
df.shape

(18249, 14)

In [7]:
df.set_index(df.columns[0], inplace=True)

In [8]:
df.dtypes

Unnamed: 0        int64
Date             object
AveragePrice    float64
Total Volume    float64
4046            float64
4225            float64
4770            float64
Total Bags      float64
Small Bags      float64
Large Bags      float64
XLarge Bags     float64
type             object
year              int64
region           object
dtype: object

Some relevant columns in the dataset:

    Date - The date of the observation
    AveragePrice - the average price of a single avocado
    type - conventional or organic
    year - the year
    Region - the city or region of the observation
    Total Volume - Total number of avocados sold
    4046 - Total number of avocados with PLU 4046 sold
    4225 - Total number of avocados with PLU 4225 sold
    4770 - Total number of avocados with PLU 4770 sold
